plugins/modules/aws_kms.py

```diff
@@ -863,6 +904,10 @@ def update_key(connection, module, key):
     changed |= update_grants(connection, module, key, module.params.get('grants'), module.params.get('purge_grants'))
     changed |= update_key_rotation(connection, module, key, module.params.get('enable_key_rotation'))
 
+    # Pause to wait for updates
+    if changed and not module.check_mode:
+        sleep(5)
```

Comment:
I'm not a fan of using sleep here because it isn't probably solving the issue. If sometimes the update operation will take more than 5 seconds we will be in the same situation. Maybe @tremble can suggest a better approach instead.

Reply:
I think what we'd need to do here is a build a comparison function that constructs what we expect to returned by `get_key_details()`, and checks to see if `result` matches (or at least the relevant keys).  If it doesn't, retry (some limited number of times).

Or else change every update/ensure function to do something similar - "connection.update_key_description, then connection.describe_key and check the description".  I don't like this option much though.

Reply:
> I think what we'd need to do here is a build a comparison function that constructs what we expect to returned by get_key_details(), and checks to see if result matches (or at least the relevant keys). If it doesn't, retry (some limited number of times).

`sleep`  is used like this in [s3_lifecycle](https://github.com/ansible-collections/community.aws/blob/main/plugins/modules/s3_lifecycle.py#L486-L494)

Reply:
@tremble - tagging you here because this is simliar to the igw issue where on one call of getting the key, description is _new-description_, then on following call, description is _old-description_. same with tags, policies, etc

Reply:
@jatorcasso Gotcha.  In that case the sleep sounds justified, can you please update the code comment to briefly explain that the API stays inconsistent too long to do a simple check and we need to wait to be confident the change has propagated?  Otherwise we might try to change it later and run into this problem again.

Reply:
To summarize the OT (Slack) conversation around this - it seems that all of the proposed workarounds regarding the wait period after updates and inconsistencies in whats being reflected when fetching key metadata don't fix the problem (integration tests still failing on idempotency tests). Documenting this in the module as well as adding waits in integration tests was decided to be the best solution (albeit not a great one).